assets/src/stories-editor/components/resizable-box/index.js

```diff
@@ -215,40 +249,84 @@ class EnhancedResizableBox extends Component {
 						lastDeltaW = deltaW;
 					}
 
-					if ( ! angle ) {
-						// If the resizing is to left or top then we have to compensate
-						if ( REVERSE_WIDTH_CALCULATIONS.includes( direction ) ) {
-							const leftInPx = getPixelsFromPercentage( 'x', parseFloat( blockElementLeft ) );
-							blockElement.style.left = getPercentageFromPixels( 'x', leftInPx - lastDeltaW ) + '%';
+					const dimensions = blockElement.getBoundingClientRect();
+
+					// We calculate with the block's actual dimensions relative to the page it's on.
+					let {
+						top: actualTop,
+						right: actualRight,
+						bottom: actualBottom,
+						left: actualLeft,
+					} = dimensions;
+
+					actualTop -= parentBlockOffsetTop;
+					actualRight -= parentBlockOffsetLeft;
+					actualBottom -= parentBlockOffsetTop;
+					actualLeft -= parentBlockOffsetLeft;
+
+					const horizontalCenter = actualLeft + ( ( actualRight - actualLeft ) / 2 );
+					const verticalCenter = actualTop + ( ( actualBottom - actualTop ) / 2 );
+
+					const snappingEnabled = ! event.getModifierState( 'Alt' );
+
+					if ( snappingEnabled ) {
```

Comment:
Could this be moved to a helper instead since it seems to be an exact duplicate of the logic of populating the `newSnapLines` in `draggable.js`? Perhaps there was a reason to keep it separately in both, mentioning just in case anyway.

Reply:
It was originally separated because this section included also the actual snapping (which was different for the two scenarios), and not just adding the guidelines to the array. But I guess this can be just as well done _after_ this section.